snapshot/git/gitfiler/setup.go

```diff
@@ -0,0 +1,52 @@
+package gitfiler
+
+import (
+	"io/ioutil"
+	"path"
+
+	"github.com/scootdev/scoot/os/temp"
+	"github.com/scootdev/scoot/snapshot/git/repo"
+)
+
+// Utilities for creating Checkouters.
+
+// RepoGetter lets a client get a Repository to use as a Reference Repository.
+type RepoGetter interface {
+	// Get gets the Repository to use as a reference repository.
+	Get() (*repo.Repository, error)
+}
+
+// A Pool that will only have a signle repo, populated by repoGetter, that serves as long as doneCh
+func NewSingleRepoPool(repoGetter RepoGetter, doneCh chan struct{}) *RepoPool {
+	singlePool := NewRepoPool(nil, nil, doneCh)
+	go func() {
+		r, err := repoGetter.Get()
+		singlePool.Release(r, err)
+	}()
+	return singlePool
+}
+
+// A Checkouter that checks out from a single repo populated by a NewSingleRepoPool)
+func NewSingleRepoCheckouter(repoGetter RepoGetter, doneCh chan struct{}) *Checkouter {
+	pool := NewSingleRepoPool(repoGetter, doneCh)
+	return NewCheckouter(pool)
+}
+
+// A Checkouter that creates a new repo with git clone --reference for each checkout
+func NewRefRepoCloningCheckouter(refRepoGetter RepoGetter, clonesDir *temp.TempDir, doneCh chan struct{}) *Checkouter {
+	refPool := NewSingleRepoPool(refRepoGetter, doneCh)
+
+	cloner := &refCloner{refPool: refPool, clonesDir: clonesDir}
+	var clones []*repo.Repository
+	fis, _ := ioutil.ReadDir(clonesDir.Dir)
```

Comment:
You're right, I was thinking could do a form of COW for the working directory but that's more magic FS territory. To clarify, much of this code goes away if/when we switch to magic FS?